src/Connection/Connection.php

```diff
@@ -72,13 +72,70 @@ public static function connectionArgs()
      */
     public static function connectionDefinitions(array $config)
     {
+        $edgeType = array_key_exists('edgeType', $config) ? $config['edgeType'] : null;
+
+        $edgeType = $edgeType ? : self::createEdgeType($config);
```

Comment:
The option to pass `edgeType` here seems redundant when we introduce the functionality to create the types individually.